pulpcore/app/viewsets/task.py

```diff
@@ -90,6 +96,12 @@ class TaskViewSet(
                 "effect": "allow",
                 "condition": "has_model_or_obj_perms:core.change_task",
             },
+            {
+                "action": ["purge"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_model_or_obj_perms:core.delete_task",
```

Comment:
There's no use to check for obj_perm here. It's a bulk endpoint.
```suggestion
                "condition": "has_model_perms:core.delete_task",
```

Reply:
Requiring this permission on the model level kind of sounds like restricting to the "tasks super admins".